UnityProjectDP/Assets/Scripts/Visualization/UI/PopUps/ParameterPopUp.cs

```diff
@@ -1,47 +1,30 @@
-using System.Collections.Generic;
-using System.Linq;
-using AnimArch.Visualization.Diagrams;
-using OALProgramControl;
 using TMPro;
-using UnityEngine.UI;
 
 namespace AnimArch.Visualization.UI
 {
-    public class ParameterPopUp : AbstractPopUp
+    public class ParameterPopUp : DropdownPopUp
     {
-        public TMP_Dropdown dropdown;
-        private readonly HashSet<TMP_Dropdown.OptionData> _variableData = new();
         public TMP_Text confirm;
         private string _formerParam;
-        
-        private void UpdateDropdown()
-        {
-            var classNames = DiagramPool.Instance.ClassDiagram.GetClassList().Select(x => x.Name);
-            
-            dropdown.options.RemoveAll(x => _variableData.Contains(x));
-            _variableData.Clear();
-            _variableData.UnionWith(classNames.Select(x => new TMP_Dropdown.OptionData(x)));
-            dropdown.options.AddRange(_variableData);
-        }
-
 
         public void ActivateCreation()
         {
             panel.SetActive(true);
-            UpdateDropdown();
             confirm.text = "Add";
+            UpdateDropdown();
```

Comment:
Tento update tu je zamerne miesto base.ActivateCreation?

Reply:
ParameterPopUp nepoužíva classText ako ostatné PopUpy, ani ho nemá zapamätaný v triede ktorá spravuje parametre (ParameterManager). classText sa však využíva pri volaní base.ActivateCreation. Prišlo mi to jednoduchšie riešenie ako prekopať AbstractPopUp alebo upraviť ParameterManager, i keď nejde o najčistejšie riešenie